Autopilot/Src/comms.cpp

```diff
@@ -0,0 +1,136 @@
+#include "comms.hpp"
+#include "usart.h"
+
+uint8_t pogiData::getPMState() {
+    return this->statusDisplay & 00001111;
+}
+
+uint8_t pogiData::getControlState() {
+    return (this->statusDisplay & 00110000) >> 4;
+}
+
+bool pogiData::getArmState() {
+     return (this->statusDisplay & 01000000) >> 6;
+}
+
+void pogiData::setArmState(bool arm) {
+    this->statusDisplay = (this->statusDisplay & 00111111) | (arm << 6);
+}
+
+void pogiData::setControlState(uint8_t state) {
+    this->statusDisplay = (this->statusDisplay & 01001111) | ((state & 00000011) << 4);
+}
+
+void pogiData::setPMState(uint8_t state) {
+    this->statusDisplay = (this->statusDisplay & 01110000) | (state & 00001111);
+}
+
+Comms* Comms::instance = nullptr;
+
+Comms::Comms() {
+    //constructor
+}
+
+Comms* Comms::GetInstance() {
+    if(!instance) {
+        instance = new Comms;
+    }
+    return instance;
+}
+
+uint8_t calculateChecksum(uint8_t* data, uint8_t start, uint8_t end) {
+    uint8_t checksum = 0;
+    for (uint8_t i = start; i < end; i++) {
+        checksum += data[i];
+    }
+    return checksum;
+}
+
+// this is a weird dumb no good way to do this, but I am tired and this works
+// manually unwrapping the class because a cast seems to not work (internet says that classes have padding)
+// will work on a v2 using only structs to get rid of those issues
+void Comms::transmitMessage() {
+    transmitFrame frame;
+    frame.payload = this->tx;
+    frame.id = 1;
+
+    uint8_t rawData[40] = { 0 };
+    uint8_t checksum = 0xFF;
+
+    rawData[0] = frame.startDelim;
```

Comment:
are you basically going through your transmit frame, putting it into an array of uint8_t, and transmitting that instead of the whole struct?